packit_service/worker/jobs.py

```diff
@@ -282,24 +288,52 @@ def process_jobs(self, event: Event) -> List[TaskResults]:
                     )
                 return processing_results
 
+            signatures = []
+
             # we want to run handlers for all possible jobs, not just the first one
-            signatures = [
-                handler_kls.get_signature(event=event, job=job_config)
-                for job_config in job_configs
-            ]
+            for job_config in job_configs:
+                handler = handler_kls(
+                    package_config=event.package_config,
+                    job_config=job_config,
+                    event=event.get_dict(),
+                )
+                if not handler.pre_check():
+                    continue
+
+                if isinstance(handler, (CoprBuildHandler, KojiBuildHandler)):
+                    helper = (
+                        CoprBuildJobHelper
+                        if isinstance(handler, CoprBuildHandler)
+                        else KojiBuildJobHelper
+                    )
+                    job_helper = helper(
+                        service_config=self.service_config,
+                        package_config=event.package_config,
+                        project=event.project,
+                        metadata=EventData.from_event_dict(event.get_dict()),
+                        db_trigger=event.db_trigger,
+                        job_config=job_config,
+                    )
+
+                    job_helper.report_status_to_all(
+                        description=TASK_ACCEPTED,
+                        state=CommitStatus.pending,
+                        url="",
+                    )
+                signatures.append(
+                    handler_kls.get_signature(event=event, job=job_config)
+                )
```

Comment:
I'm wondering if we could create a celery task chain here: https://docs.celeryproject.org/en/stable/userguide/canvas.html#chains so we can reduce the size of this method

No need to do any changes here right now, just a food for thought.